code/game/objects/items/control_wand.dm

```diff
@@ -55,7 +58,7 @@
 				else
 					D.emergency = 1
 				D.update_icon()
-	else
+	else if(!(D.is_special)) //so we don't get double messages for special doors
```

Comment:
Because you checked `if(D.is_special)` above, and had a `return` statement within that. So this is guaranteed to be true.